server/src/main/java/org/opensearch/throttling/admissioncontrol/controllers/IOBasedAdmissionController.java

```diff
@@ -0,0 +1,139 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.throttling.admissioncontrol.controllers;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.opensearch.common.settings.ClusterSettings;
+import org.opensearch.common.settings.Settings;
+import org.opensearch.throttling.admissioncontrol.enums.AdmissionControlMode;
+import org.opensearch.throttling.admissioncontrol.settings.IOBasedAdmissionControllerSettings;
+import org.opensearch.throttling.admissioncontrol.utils.AdmissionControlActionsMap;
+
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.concurrent.atomic.AtomicLong;
+
+/**
+ *  Class for IO Based Admission Controller in OpenSearch, which aims to provide IO utilisation admission control.
+ *  It provides methods to apply admission control if configured limit has been reached
+ */
+public class IOBasedAdmissionController implements AdmissionController {
+    private static final Logger LOGGER = LogManager.getLogger(IOBasedAdmissionController.class);
+    private final String admissionControllerName;
+    public IOBasedAdmissionControllerSettings settings;
+
+    private final Set<String> allowedTransportActions;
```

Comment:
"allowedTransportActions" sounds like we allow these actions to pass through. can we rename to 'admissionControlTransportActionsList' or something similar ?